Source/control.cpp

```diff
@@ -1180,22 +1172,11 @@ void CheckPanelInfo()
 			case RSPLTYPE_SCROLL: {
 				strcpy(tempstr, fmt::format(_("Scroll of {:s}"), _(spelldata[v].sNameText)).c_str());
 				AddPanelString(tempstr);
-				int s = 0;
-				for (int i = 0; i < myPlayer._pNumInv; i++) {
-					if (!myPlayer.InvList[i].isEmpty()
-					    && (myPlayer.InvList[i]._iMiscId == IMISC_SCROLL || myPlayer.InvList[i]._iMiscId == IMISC_SCROLLT)
-					    && myPlayer.InvList[i]._iSpell == v) {
-						s++;
-					}
-				}
-				for (auto &item : myPlayer.SpdList) {
-					if (!item.isEmpty()
-					    && (item._iMiscId == IMISC_SCROLL || item._iMiscId == IMISC_SCROLLT)
-					    && item._iSpell == v) {
-						s++;
-					}
-				}
-				strcpy(tempstr, fmt::format(ngettext("{:d} Scroll", "{:d} Scrolls", s), s).c_str());
+				const InventoryAndBeltPlayerItemsRange items { myPlayer };
+				const int count = std::count_if(items.begin(), items.end(), [v](const Item &item) {
+					return item.IsScrollOf(v);

```

Comment:
Can we rename `v` to `spellId` as well so that the changes in this PR are consistent and readable?

Reply:
Done